include/deal.II/matrix_free/tensor_product_point_kernels.h

```diff
@@ -75,6 +75,32 @@ namespace internal
 
 
 
+  template <int dim, typename Number>
+  inline void
+  compute_values_of_array_in_pairs(
+    dealii::ndarray<Number, 2, dim>                    *shapes,
+    const std::vector<Polynomials::Polynomial<double>> &poly,
+    const Point<dim, Number>                           *p)
```

Comment:
I am not sure if this was discussed in previous reviews, but are raw pointers here necessary? In this case could the function just take a const reference to `p1` and another one to `p2`? That way there would be no danger of someone handing over a pointer to an array that doesnt have 2 entries.

Reply:
No, the raw pointers should not be necessary here. I checked with my compiler and what I see is two more integer additions to pass over the second reference (in an additional register that is part of the calling convention on my x86-64 system), but I think that is acceptable for the increase in readability.